src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogHttpClient.java

```diff
@@ -55,8 +74,39 @@ public boolean sendEvent(JSONObject payload) {
         return status;
     }
 
+    @Override
+    public void incrementCounter(String name, String hostname, JSONArray tags) {
+        ConcurrentMap<CounterMetric, Integer> counters = ConcurrentMetricCounters.Counters.get();
+        CounterMetric counterMetric = new CounterMetric(tags, name, hostname);
+        Integer previousValue = counters.putIfAbsent(counterMetric, 1);
+        if (previousValue != null){
+            boolean ok = counters.replace(counterMetric, previousValue, previousValue + 1);
+            while(!ok) {
+                previousValue = counters.get(counterMetric);
+                ok = counters.replace(counterMetric, previousValue, previousValue + 1);
+            }
+        }
+    }
+
+    @Override
+    public void flushCounters() {
+        ConcurrentMap<CounterMetric, Integer> counters = ConcurrentMetricCounters.Counters.get();
+        ConcurrentMetricCounters.Counters.set(new ConcurrentHashMap<CounterMetric, Integer>());
+
+        // Submit all metrics as gauge
+        for (CounterMetric counterMetric: counters.keySet()) {
+            int count = counters.get(counterMetric);
+
+            this.postMetric(counterMetric.getMetricName(), count, counterMetric.getHostname(), counterMetric.getTags(), "rate");
```

Comment:
Why send as rate and not as count ?

Reply:
I guess backward compatibility with what dogstatsd was doing before ?

Reply:
I should have read this feedback sooner... Would have prevented some long debugging